api/src/main/java/org/openmrs/Cohort.java

```diff
@@ -208,15 +210,29 @@ public boolean removeMembership(CohortMembership cohortMembership) {
 	public Collection<CohortMembership> getActiveMemberships() {
 		return getActiveMemberships(new Date());
 	}
-	
+
 	/**
+	 * Searches for the {@code patients} membership and returns it.
+	 * <p>
+	 * If the patient has no membership, an empty Optional is returned.
+	 *
+	 * @param patient Holder of the searched membership.
+	 * @return Optional membership of the given patient.
+	 * @throws NoSuchElementException Thrown if the patient has no active membership.
 	 * @since 2.1.0
 	 */
 	public CohortMembership getActiveMembership(Patient patient) {
-		return getMemberships().stream().filter(m -> m.isActive() && m.getPatientId().equals(patient.getPatientId()))
-		        .collect(Collectors.toList()).get(0);
+		Predicate<CohortMembership> patientMatcher =
+			membership -> membership.getPatientId().equals(patient.getPatientId());
```

Comment:
Just for minimal and efficient code, why don't we do all this in the `#filter()`? With that we could get rid of the above.

Reply:
The line is actually somewhat longer and according to Joshua Blochs Effective Java, its a better practice to do it as I wrote.